src/coreclr/tools/cdac-build-tool/ObjectFileScraper.cs

```diff
@@ -0,0 +1,497 @@
+        public byte GetByte(long offset) => Data.Span[(int)offset];
+
+
+        public void GetBytes(long offset, Span<byte> buffer) => Data.Span.Slice((int)offset, buffer.Length).CopyTo(buffer);
+
+        public void ResetPosition(long position)
+        {
+            _position = position;
+        }
+
+        public ulong ReadUInt64()
+        {
+            var value = GetUInt64(_position);
+            _position += sizeof(ulong);
+            return value;
+        }
+        public uint ReadUInt32()
+        {
+            var value = GetUInt32(_position);
+            _position += sizeof(uint);
+            return value;
+        }
+        public ushort ReadUInt16()
+        {
+            var value = GetUInt16(_position);
+            _position += sizeof(ushort);
+            return value;
+        }
+
+        public byte ReadByte()
+        {
+            var value = GetByte(_position);
+            _position += sizeof(byte);
+            return value;
+        }
+        public void ReadBytes(Span<byte> buffer)
+        {
+            GetBytes(_position, buffer);
+            _position += buffer.Length;
+        }
+
+        public void Skip(int count)
+        {
+            _position += count;
+        }
+    }
+
+
+    private struct HeaderDirectory
+    {
+        public uint FlagsAndBaselineStart;
+        public uint TypesStart;
+
+        public uint FieldPoolStart;
+        public uint GlobalLiteralValuesStart;
+
+        public uint GlobalPointersStart;
+        public uint NamesStart;
+
+        public uint TypeCount;
+        public uint FieldPoolCount;
+
+        public uint GlobalLiteralValuesCount;
+        public uint GlobalPointerValuesCount;
+
+        public uint NamesPoolCount;
+
+        public byte TypeSpecSize;
+        public byte FieldSpecSize;
+        public byte GlobalLiteralSpecSize;
+        public byte GlobalPointerSpecSize;
+    };
+
+    private static void DumpHeaderDirectory(HeaderDirectory headerDirectory)
+    {
+        Console.WriteLine($"Baseline Start = {headerDirectory.FlagsAndBaselineStart}");
+        Console.WriteLine($"Types Start = {headerDirectory.TypesStart}");
+        Console.WriteLine($"Field Pool  Start = {headerDirectory.FieldPoolStart}");
+        Console.WriteLine($"Global Literals Start = {headerDirectory.GlobalLiteralValuesStart}");
+        Console.WriteLine($"Global Pointers Start = {headerDirectory.GlobalPointersStart}");
+        Console.WriteLine($"Names Start = {headerDirectory.NamesStart}");
+        Console.WriteLine($"TypeCount = {headerDirectory.TypeCount}");
+        Console.WriteLine($"FieldPoolCount = {headerDirectory.FieldPoolCount}");
+        Console.WriteLine($"GlobalLiteralValuesCount = {headerDirectory.GlobalLiteralValuesCount}");
+        Console.WriteLine($"GlobalPointerValuesCount = {headerDirectory.GlobalPointerValuesCount}");
```

Comment:
Nit: It looks like some of these are offsets. Offset in decimal form are very hard to reason about. I think it might be easier to read as hexadecimal if they are nicely aligned it creates easily readable output.